### ISSUE TITLE
#![macro_use] is being ignored

### ISSUE THREAD

**Issue by Veykril** [rust-analyzer/rust-analyzer#7501]
```rust
mod foo;
bar!();

//- /foo.rs
#![macro_use]
macro_rules! bar { () => ()}
```
The bar invocation is unresolved.
For inline modules this does work though,
```rust

mod foo {
    #![macro_use]
    macro_rules! bar {
        () => {};
    }
}
bar!();
```
does not emit a diagnostic.

**Comment by Veykril** [comment]
I think the problem lies in the attribute query only checking the `AST`, so for non-inline modules this causes us to skip over the inner attributes, since the source file and the mod declaration are separate syntax nodes.
